### DISCUSSION TITLE
How do i get the authorizer id?

### DISCUSSION THREAD

**Discussion by wz2b** [aws/aws-cdk#24502] [Q&A]
After I create an HttpUserPoolAuthorizer how do I get its authorizerId?  I need to be able to export that ID.

**Comment by wz2b** [comment]
As a workaround I'm just breaking the private:
```
const authId = (authorizer as any)["authorizer"]["authorizerId"]
```

I feel dirty.  But I really need to export it, and I can't figure out another way to get it.

**Reply by wz2b** [reply]
Hmm at first that seemed to work but now it does not.  authorizer.authorizer is only set [after binding](https://github.com/aws/aws-cdk/blob/1b2014eef9e3f2190b2cce79c55f635cc1f167e3/packages/%40aws-cdk/aws-apigatewayv2-authorizers/lib/http/user-pool.ts#L59) so it must be that happened the first time I ran it but not the second.

**Comment by khushail** [comment]
Hi @wz2b , using[ escape hatches ](https://docs.aws.amazon.com/cdk/v2/guide/cfn_layer.html)to access the private [HttpAuthorizer](https://github.com/aws/aws-cdk/blob/ddbfac7a32fa5966b2c0911e62f96fa4b9859b5a/packages/%40aws-cdk/aws-apigatewayv2-authorizers/lib/http/user-pool.ts#L40) might be helpful here.

**Comment by jasnova-mrj** [comment]
How is a method to get authorizer ID difficult to implement in the official CDK library?
